### PR TITLE
chore(python): add docstring for limit behavior

### LINKED ISSUE #796: Document "no limit" behavior for LanceQueryBuilder
### Description

https://github.com/lancedb/lancedb/blob/121687231c2979844d3b3bc4662c848007cd1e02/python/lancedb/query.py#L263

The type annotation for `LanceQueryBuilder.limit()` says limit must be an int and there's no mention of what you should pass if you want "no limit" (it's `None` ... a user might think you need to pass in `limit=-1`?).  Type annotation should be changed to `Optional[int]` and `None` behavior should be documented accordingly.

Somewhat related - it is a little non-intuitive that the default behavior is `limit = 10` as opposed to no limit.

### Link

_No response_

### PR DESCRIPTION
**Pull request by changhiskhan**
Closes #796

### CODE REVIEW

File: python/lancedb/query.py [hunk 1]
```diff
@@ -260,20 +260,30 @@ def to_pydantic(self, model: Type[LanceModel]) -> List[LanceModel]:
...
         ----------
         limit: int
             The maximum number of results to return.
+            By default the query is limited to the first 10.
+            Call this method and pass in any negative value
+            or None to remove the limit.
+            *WARNING* if you have a large dataset, removing
+            the limit can potentially result in reading a
+            large amount of data into memory and cause
+            out of memory issues.
```
Comment (westonpace) [inline python/lancedb/query.py:276]: Even if I call `to_batches`?

File: python/lancedb/query.py [hunk 2]
```diff
@@ -260,20 +260,30 @@ def to_pydantic(self, model: Type[LanceModel]) -> List[LanceModel]:
...
 
         Returns
         -------
         LanceQueryBuilder
             The LanceQueryBuilder object.
         """
-        self._limit = limit
+        if limit is None or limit < 0:
```
Comment (westonpace) [inline python/lancedb/query.py:283]: What if `limit == 0`?

File: python/lancedb/query.py (hunk 1 above)
Comment (changhiskhan) [reply python/lancedb/query.py:276]: we don't have `to_batches` api yet. I'll create a follow-up issue

File: python/lancedb/query.py (hunk 2 above)
Comment (changhiskhan) [reply python/lancedb/query.py]: made limit == 0 also act as if there's no limit

**Pull request merged**